### PR TITLE
Imbemr 2.x

### PR DESCRIPTION
**Pull request by mseaton**

### CODE REVIEW

File: configuration/appframework/coreapps/appointments_app.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[
+  {
+    "id": "coreapps.appointmentsApp",
+    "instanceOf": "coreapps.template.findPatient",
```
Comment (mseaton) [inline configuration/appframework/coreapps/appointments_app.json]: These apps should only be instances of the coreapps.template.findPatient app if it makes sense to first have a patient search when you enter it, and then for the result of that patient search to take you to a patient-specific page.  In most of the cases I see here and below, this is not the case.  These should instead follow the model of things like the home_configuration_extension.json - eg the imbemr.myAccountHomePageLink - where you are just configuring a link on the home page to a specific page in the application, not configuring something that requires a patient search.  I understand maybe this is actually incorrect in the coreapps code that we copied, but we can rectify that here.

File: configuration/appframework/coreapps/billing_app.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[
+  {
+    "id": "coreapps.billingApp",
+    "instanceOf": "coreapps.template.findPatient",
```
Comment (mseaton) [inline configuration/appframework/coreapps/billing_app.json]: Same comment as above - this is not a findPatient app

File: configuration/appframework/coreapps/cohortBuilder_app.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[
+  {
+    "id": "coreapps.cohortBuilderApp",
+    "instanceOf": "coreapps.template.findPatient",
```
Comment (mseaton) [inline configuration/appframework/coreapps/cohortBuilder_app.json]: Same comment as above - this is not a findPatient app

File: configuration/appframework/coreapps/dictionary_app.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+[
+  {
+    "id": "coreapps.dictionaryApp",
+    "instanceOf": "coreapps.template.findPatient",
```
Comment (mseaton) [inline configuration/appframework/coreapps/dictionary_app.json]: Same comment as above - this is not a findPatient app

File: configuration/appframework/coreapps/findpatient_app.json [hunk 5]
```diff
@@ -4,12 +4,15 @@
         "instanceOf": "coreapps.template.findPatient",
         "description": "Basic patient search that goes to the patient dashboard",
         "order": 2,
+        "config": {
+            "afterSelectedUrl": "/patientDashboard.form?patientId={{patientDbId}}"
+        },
         "extensions": [
```
Comment (mseaton) [inline configuration/appframework/coreapps/findpatient_app.json:10]: This is correct.   See how this is what an instance of a findPatient app should do - it should have an afterSelectedUrl that links to a patient-specific page

File: configuration/appframework/referenceapplication/visitActions_extension.json [hunk 6]
```diff
@@ -52,13 +52,32 @@
         "extensionPointId": "patientDashboard.visitActions",
         "type": "link",
         "label": "referenceapplication.app.capturevitals.title",
-        "url": "htmlformentryui/htmlform/enterHtmlFormWithSimpleUi.page?patientId={{patient.uuid}}&visitId={{visit.uuid}}&definitionUiResource=referenceapplication:htmlforms/vitals.xml",
+        "url": "htmlformentryui/htmlform/enterHtmlFormWithSimpleUi.page?patientId={{patient.uuid}}&visitId={{visit.uuid}}&definitionUiResource=file%3Aconfiguration%2Fhtmlforms%2Fvitals.xml",
```
Comment (mseaton) [inline configuration/appframework/referenceapplication/visitActions_extension.json]: I don't think the `%3A` and `%2F` syntax is needed, and it's harder to read.  Let's use `:` and `/` unless proven that we can't.

File: configuration/appframework/referenceapplication/visitActions_extension.json [hunk 7]
```diff
@@ -52,13 +52,32 @@
...
         "requiredPrivilege": "Task: referenceapplication.vitals",
         "require": "visit.active"
     },
-
+    {
+        "id": "referenceapplication.realTime.admission",
+        "extensionPointId": "patientDashboard.visitActions",
+        "type": "link",
+        "label": "Admission Form",
+        "url": "htmlformentryui/htmlform/enterHtmlFormWithSimpleUi.page?patientId={{patient.uuid}}&visitId={{visit.uuid}}&definitionUiResource=file%3Aconfiguration%2Fhtmlforms%2Fauto-ipd-admission.xml",
```
Comment (mseaton) [inline configuration/appframework/referenceapplication/visitActions_extension.json]: Same here with the encoded text.  Let's not pre-encode, but assume that the page rendering will encode correctly.

File: configuration/appframework/referenceapplication/visitActions_extension.json [hunk 8]
```diff
@@ -52,13 +52,32 @@
...
+        "icon": "icon-stethoscope",
+        "order": 52,
+        "require": "visit.active && visit.admitted"
+    },
+    {
+        "id": "referenceapplication.realTime.labOrder",
+        "extensionPointId": "patientDashboard.visitActions",
+        "type": "link",
+        "label": "Lab Request",
+        "url": "htmlformentryui/htmlform/enterHtmlFormWithSimpleUi.page?patientId={{patient.uuid}}&visitId={{visit.uuid}}&definitionUiResource=file%3Aconfiguration%2Fhtmlforms%2Fauto-ipd-admission.xml",
```
Comment (mseaton) [inline configuration/appframework/referenceapplication/visitActions_extension.json]: Ditto

File: configuration/appframework/registrationapp/editPatientBillingInfo_extension.json [hunk 9]
```diff
@@ -0,0 +1,9 @@
+[
+    {
+        "id": "registrationapp.editPatientBillingInfo",
+        "extensionPointId": "patientHeader.editPatientBillingInfo",
+        "type": "link",
+        "label": "general.edit",
+        "url": "registrationapp/editSection.page?patientId={{patient.patientId}}&sectionId=billingInfo&appId=referenceapplication.registrationapp.registerPatient"
```
Comment (mseaton) [inline configuration/appframework/registrationapp/editPatientBillingInfo_extension.json]: Makes sense, though we don't actually have a billingInfo section in the registration app yet, do we?

File: configuration/metadatatermmappings/metadataterms.csv [hunk 10]
```diff
@@ -0,0 +1,7 @@
+Uuid,Void/Retire,Mapping code,Mapping source,Metadata class name,Metadata Uuid,_order:2000
```
Comment (mseaton) [inline configuration/metadatatermmappings/metadataterms.csv]: Defer to you as to whether all of these are correct and accurate.

**Review by mseaton — COMMENTED** [review]
See my comments @nic10ben .  Overall looks good.

File: configuration/appframework/coreapps/appointments_app.json (hunk 1 above)
Comment (nic10ben) [reply configuration/appframework/coreapps/appointments_app.json]: Ohh I understand that, I will change it. 
I think I made the same error on all apps that links to legacy UI that I created.
Thank you!

File: configuration/appframework/coreapps/findpatient_app.json (hunk 5 above)
Comment (nic10ben) [reply configuration/appframework/coreapps/findpatient_app.json:10]: Understood, thank you so much!

File: configuration/appframework/referenceapplication/visitActions_extension.json (hunk 6 above)
Comment (nic10ben) [reply configuration/appframework/referenceapplication/visitActions_extension.json]: I think : and / refused to work but We really don't need this change to be used, I just did that for me to make sure that we can access html form from Overall actions or Active actions.

I will try to use : and / to see if it really can't work.

Thank you so much again, Mike.

File: configuration/appframework/registrationapp/editPatientBillingInfo_extension.json (hunk 9 above)
Comment (nic10ben) [reply configuration/appframework/registrationapp/editPatientBillingInfo_extension.json]: We don't for now, just did it when I was exploring extensions, I should have removed that before commit. Sorry for that, Thanks!

File: configuration/metadatatermmappings/metadataterms.csv (hunk 10 above)
Comment (nic10ben) [reply configuration/metadatatermmappings/metadataterms.csv]: Yes, they are correct, I used them from our DB.

**Pull request merged**